grid-demo/src/app/angular/column-ordering/custom-clr-column-ordering-column.directive.ts

```diff
@@ -1,44 +1,71 @@
 import { CdkDrag } from '@angular/cdk/drag-drop';
-import { Directive, ElementRef, HostBinding, HostListener, Input } from '@angular/core';
-import { ClrDatagrid } from '@clr/angular';
+import {
+  ChangeDetectorRef,
+  Directive,
+  ElementRef,
+  EventEmitter,
+  HostBinding,
+  HostListener,
+  Input,
+  OnDestroy,
+  OnInit,
+  Output,
+} from '@angular/core';
+import { ClrDatagrid, ClrDatagridColumn } from '@clr/angular';
+import { Subscription } from 'rxjs';
 import { CustomClrColumnOrderingGridDirective } from './custom-clr-column-ordering-grid.directive';
+import { CustomClrColumnOrderingService } from './custom-clr-column-ordering.service';
 
 @Directive({
   selector: 'clr-dg-column[customClrColumnOrderingColumn]',
 })
-export class CustomClrColumnOrderingColumnDirective {
+export class CustomClrColumnOrderingColumnDirective implements OnInit, OnDestroy {
   constructor(
     private readonly gridDirective: CustomClrColumnOrderingGridDirective,
     private readonly datagrid: ClrDatagrid,
     private readonly elementRef: ElementRef,
+    private readonly columnOrderingService: CustomClrColumnOrderingService,
+    private readonly changeDetectorRef: ChangeDetectorRef,
+    private readonly column: ClrDatagridColumn,
     cdkDrag: CdkDrag
   ) {
     cdkDrag.previewContainer = 'parent';
   }
 
+  private subscription: Subscription | null = null;
+
   @Input() customClrColumnOrderingColumn: any;
   @Input() columnIndex!: number;
+  @Output() columnPickedUp = new EventEmitter<void>();
+  @Output() columnDropped = new EventEmitter<void>();
+  _isGrabbed = false;
```

Comment:
This should be removed. Rebasing should take care of this.

Reply:
fixed